### DISCUSSION TITLE
TrySetResultAsync missing in 120 version, how to handle this?

### DISCUSSION THREAD

**Discussion by lofti198** [cefsharp/CefSharp#4660] [Q&A] [answered]
Windows 10, .NET framework 4.8, updated from 91 to 120 version and getting this error:
![image](https://github.com/cefsharp/CefSharp/assets/42960488/beccfdf0-3971-456a-815e-a95033ba06f8)

Error	CS1061	'TaskCompletionSource<bool>' does not contain a definition for 'TrySetResultAsync' and no accessible extension method 'TrySetResultAsync' accepting a first argument of type 'TaskCompletionSource<bool>' could be found (are you missing a using directive or an assembly reference?)

What is the best way to handle this?

**Comment by campersau** [comment, marked as answer]
The extension method got removed with the upgrade to .NET Framework 4.6.2 here:
https://github.com/cefsharp/CefSharp/commit/1def8a7937dce48e6f6264ae147cd63bdfd8c598#diff-fab8d4eeceb0db5072815c35d8efa6886d14f4529f7ce4d9d228682422e492c4L95-L108

You can create a `TaskCompletionSource` with `TaskCreationOptions.RunContinuationsAsynchronously` option and just use `TrySetResult`.
```c#
var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);

tcs.TrySetResult(true);
```

**Comment by lofti198** [comment]
Works awesom!
